tasks/components/OrganizationMemberList.tsx

```diff
@@ -0,0 +1,257 @@
+import { useState } from 'react'
+import {
+  useReactTable,
+  createColumnHelper,
+  getCoreRowModel,
+  getPaginationRowModel
+} from '@tanstack/react-table'
+import { tw } from '@helpwave/common/twind/index'
+import type { Languages } from '@helpwave/common/hooks/useLanguage'
+import type { PropsWithLanguage } from '@helpwave/common/hooks/useTranslation'
+import { useTranslation } from '@helpwave/common/hooks/useTranslation'
+import Dropdown from '../icons/TriangleDown'
+import { Pagination } from './Pagination'
+import { ProfilePicture } from './ProfilePicture'
+import { TriStateCheckbox } from './TriStateCheckbox'
+import { Button } from './Button'
+
+// TODO replace later
+export const enum Role {
+  user,
+  admin,
+}
+
+type OrganizationMemberListTranslation = {
+  edit: string,
+  remove: string,
+  deselectAll: string,
+  selectAll: string,
+  members: string,
+  addMember: string,
+  saveChanges: string,
+  role: string,
+  roleTypes: Record<Role, string>
+}
+
+const defaultOrganizationMemberListTranslations: Record<Languages, OrganizationMemberListTranslation> = {
+  en: {
+    edit: 'Edit',
+    remove: 'Remove',
+    deselectAll: 'Deselect all',
+    selectAll: 'Select all',
+    members: 'Members',
+    addMember: 'Add member',
+    saveChanges: 'Save changes',
+    role: 'Role',
+    roleTypes: { [Role.admin]: 'Admin', [Role.user]: 'User' }
+  },
+  de: {
+    edit: 'Bearbeiten',
+    remove: 'Entfernen',
+    deselectAll: 'Auswahl aufheben',
+    selectAll: 'Alle auswählen',
+    members: 'Mitglieder',
+    addMember: 'Miglied hinzufügen',
+    saveChanges: 'Speichern',
+    role: 'Rolle',
+    roleTypes: { [Role.admin]: 'Administrator', [Role.user]: 'Nutzer' }
+  }
+}
+
+type OrgMember = {
+  email: string,
+  name: string,
+  avatarURL: string,
+  role: Role
+}
+
+type OrganizationDTO = {
+  name: string,
+  members: OrgMember[]
+}
+
+export type OrganizationMemberListProps = {
+  organization: OrganizationDTO,
+  usersPerPage?: number
+}
+
+const columnHelper = createColumnHelper<OrgMember>()
+
+const columns = [
+  columnHelper.display({
+    id: 'select',
+  }),
+  columnHelper.accessor('name', {
+    id: 'name',
+  }),
+  columnHelper.accessor('role', {
+    id: 'role',
+  }),
+  columnHelper.display({
+    id: 'remove',
+  }),
+]
+
+export const OrganizationMemberList = ({
+  language,
+  usersPerPage = 5,
+  organization
+}: PropsWithLanguage<OrganizationMemberListTranslation, OrganizationMemberListProps>) => {
+  const translation = useTranslation(language, defaultOrganizationMemberListTranslations)
+  // State copy of the members, so discarding changes is possible
+  const [members, setMembers] = useState(organization.members.map((value) => {
```

Comment:
why are you doing the `.map` here?

Reply:
To create a copy of the member list, so the status quo can be maintained, if changes are not saved or discarded and to not modify the original objects.

If you have a better way of create copys please let me know

Reply:
you don't need to copy it, the original object is never modified

Reply:
If we were to allow the ability to change roles the object would be modified or does the table itself create a copy, but I will remove the mapping for now

Reply:
The object reference you pass into `useState` is never written to, because you always use `setMembers` to update the state.

Reply:
The only concern with regards to references and react is "cache invalidation"
You never modify the reference you receive, that just doesn't happen with react and is therefore no concern.

What is however a concern is wrongful invalidation and subsequent re-renders which don't need to happen